### ISSUE TITLE
Add greater accessibility support

### ISSUE THREAD

**Issue by jakezatecky** [jakezatecky/react-checkbox-tree#21]
Icon-only buttons are not readily consumable by screen readers. There should be alternative text instead or `aria-label` should be used. Click handlers on `<span>` elements are also not semantic and `<label>`s without targets may also hinder accessibility.

**Comment by jakezatecky** [comment]
Also addressed in 5198501a559cd53896da090f8b730d285e5ecf34.
